project-reports/2024/2024-annual-Hyperledger-Iroha.md

```diff
@@ -0,0 +1,100 @@
+
+# Goals
+
+## Performance Against Prior Goals
+
+The development of the Iroha 2.0 MVP is ongoing, with the targeted release at the end of March 2024 in line with the agreed-upon product roadmap.
+
+Given this is the first Hyperledger Iroha Annual Report, there is no 2022 Annual Report to look back on. Instead, this section highlights the project's accomplishments in no particular order.
+
+**New features**:
+
+* Soft fork recovery
+* Runtime-definable permissions
+* Executor, a smart contract that functions as the backbone of Iroha 2
+* Instruction visitor
+* The server-side cursor for query batching
+* `WSV` snapshots for fast restart without the block replay
+* `WASM` permission validators
+
+**Refactoring**:
+
+* The domains now have owners
+* The layout of the Iroha 2 schema was simplified (every entry is unique)
+* The [`iroha_crypto`](https://github.com/hyperledger/iroha/tree/iroha2-dev/crypto) crate superseded the Hyperledger Ursa dependency
+* Our team standardised the block layout
+* Our team standardised the transaction layout
+* Our team simplified the data model API and removed the expressions
+* Kagami was refactored and split into `iroha_swarm` and `iroha_wasm_builder_cli`
+
+Our team actively pursues enhancing community engagement, promoting Hyperledger Iroha 2 into the broader IT community, and collaborating with the Hyperledger PR team.
+
+Despite facing challenges related to a shortage of high-quality and available development resources, the team remains dedicated to the original project direction. As we work towards the MVP release, addressing challenges and seeking additional support will be pivotal for the project's success.
+
+As stated above, this quarter marks a significant evolution in the development of Iroha cryptographic infrastructure: the complete removal of Ursa's dependency from Iroha 2, the integration of its code to the [`iroha-crypto`](https://github.com/hyperledger/iroha/tree/iroha2-dev/crypto) package and the further refactoring of the cryptographic tooling.
+
+Initially, Iroha 2 used Ursa as a core cryptographic library to leverage its secure and reusable features. In turn, Ursa worked as an abstraction over other cryptography libraries. Iroha 2 benefited from Ursa's high cryptographic security standards. This change also minimised the need for redundant cryptographic code.
+
+At this point, the necessary parts of Ursa are detached and fully integrated. The refactoring of the [`iroha-crypto`](https://github.com/hyperledger/iroha/tree/iroha2-dev/crypto) library itself continues.
+
+The latest topic in Iroha's cryptographic infrastructure   addressed by our team is [#4181](https://github.com/hyperledger/iroha/pull/4181); it allows the parsed keys to be stored instead of the raw bytes.
```

Comment:
```suggestion
The latest topic in Iroha's cryptographic infrastructure addressed by our team is [#4181](https://github.com/hyperledger/iroha/pull/4181); it allows the parsed keys to be stored instead of the raw bytes.
```